agent/command/git.go

```diff
@@ -393,240 +434,260 @@ func (c *gitFetchProject) Execute(ctx context.Context, comm client.Communicator,
+	td client.TaskData) ([]string, error) {
+
+	logger.Execution().Info("Fetching additional patches.")
+	additionalPatches, err := comm.GetAdditionalPatches(ctx, conf.Task.Version, td)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get additional patches")
+	}
+	return additionalPatches, nil
+}
+
+func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
+	conf *internal.TaskConfig,
+	logger client.LoggerProducer,
+	jpm jasper.Manager,
+	projectMethod string,
+	projectToken string,
+	p *patch.Patch,
+	moduleName string) error {
 
 	var err error
-	// expand the github parameters before running the task
-	if err = util.ExpandValues(c, conf.Expansions); err != nil {
-		return errors.Wrap(err, "error expanding github parameters")
+	logger.Execution().Infof("Fetching module: %s", moduleName)
+
+	var module *model.Module
+	module, err = conf.Project.GetModuleByName(moduleName)
+	if err != nil {
+		return errors.Errorf("Couldn't get module %s: %v", moduleName, err)
+	}
+	if module == nil {
+		return errors.Errorf("No module found for %s", moduleName)
 	}
 
-	var projectMethod string
-	var projectToken string
-	projectMethod, projectToken, err = getProjectMethodAndToken(c.Token, conf.Expansions.Get(evergreen.GlobalGitHubTokenExpansion), conf.Distro.CloneMethod)
+	moduleBase := filepath.ToSlash(filepath.Join(expandModulePrefix(conf, module.Name, module.Prefix, logger), module.Name))
+
+	var revision string
+	// use submodule revisions based on the main patch. If there is a need in the future,
+	// this could maybe use the most recent submodule revision of all requested patches.
+	// We ignore set-module changes for commit queue, since we should verify HEAD before merging.
+	if p != nil {
+		patchModule := p.FindModule(moduleName)
+		if patchModule != nil {
+			if conf.Task.Requester == evergreen.MergeTestRequester {
+				revision = module.Branch
+				c.logModuleRevision(logger, revision, moduleName, "defaulting to HEAD for merge")
+			} else {
+				revision = patchModule.Githash
+				if revision != "" {
+					c.logModuleRevision(logger, revision, moduleName, "specified in set-module")
+				}
+			}
+		}
+	}
+	if revision == "" {
+		revision = c.Revisions[moduleName]
+		if revision != "" {
+			c.logModuleRevision(logger, revision, moduleName, "specified as parameter to git.get_project")
+		}
+	}
+	if revision == "" {
+		revision = conf.Expansions.Get(moduleRevExpansionName(moduleName))
+		if revision != "" {
+			c.logModuleRevision(logger, revision, moduleName, "from manifest")
+		}
+	}
+	// if there is no revision, then use the revision from the module, then branch name
+	if revision == "" {
+		if module.Ref != "" {
+			revision = module.Ref
+			c.logModuleRevision(logger, revision, moduleName, "ref field in config file")
+		} else {
+			revision = module.Branch
+			c.logModuleRevision(logger, revision, moduleName, "branch field in config file")
+		}
+	}
+	var owner, repo string
+	owner, repo, err = thirdparty.ParseGitUrl(module.Repo)
```

Comment:
Right now if owner/repo is empty we continue. Do we still want to validate this?

Reply:
I added this to `ParseGitUrl`